src/compile/axis/rules.ts

```diff
@@ -53,6 +53,24 @@ export function labelOverlap(fieldDef: FieldDef<string>, specifiedAxis: Axis, ch
   return undefined;
 }
 
+export function minMaxExtent(specifiedExtent: number, isGridAxis: boolean, scaleType: ScaleType) {
+  if (!isGridAxis) {
+    // Only apply extent to non-grid axes as grid axes do not have labels and title.
+    // Otherwise, we would add unnecessary extent to faceted plots.
+    if (specifiedExtent) {
+      return specifiedExtent;
+    }
+
+    // For quantitative scale, set extent to 25 by default to avoid jumpy axis title
+    // (Fix https://github.com/vega/vega-lite/issues/2282)
+    if (getScaleCategory(scaleType) === 'numeric') {
+      return 25;
```

Comment:
> Why 25px? In my tests with zooming quantitative scales (e.g., the panzoom_scatter or the Gapminder bubble_health_income examples), with 25px extents, axis labels with one decimal place were flush against the title and with two decimal places overlapped it.
> 30px seems to give us more breathing room (1dp still maintains padding while 2dps are flush).

Jeff actually originally suggests 40px but I felt like that's a bit too big so I decrease it (maybe I overdo it.)  I could change to `30` but wonder if you like 40 better also.  

> Also, rather than hardcoding this, should we move it to a config value and/or 

I'll add `config.axis.quantitativeExtent`

> something that can be customized by the user in their specs? E.g., encoding.x.axis.extent = [min, max]?

Users can already do `encoding.x.axis.max/minExtent`.  (This function only determines default rules when not specified.)